projects/Mallard/src/navigation/helpers.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react'
+import {
+    NavigationScreenProp,
+    NavigationNavigatorProps,
+} from 'react-navigation'
+import { NavigationContainerProps } from 'react-navigation'
+import { NavigationContainer } from 'react-navigation'
+
+/**
+ *
+ * @param Component - component that doesn't want to have navigation as a dependency
+ * @param mapper - function to generate props from navigation
+ *
+ * Much like `mapDispatchToProps` in `redux`. Means we can decouple out components from navigation.
+ */
+const mapNavigationToProps = <T extends {}, P extends {}>(
```

Comment:
what does this pattern do for us that we couldn't get using the `withNavigation` HOC?

Reply:
It's just a habit of not trying to leak implementations into child components. I'd rather have the component depend on a simple `onComplete` callback rather than "knowing" how to call the next route (and what the next route should be). So the parent (defined right next to all the routes) can define that and pass it (as seen here): https://github.com/guardian/editions/pull/116/files#diff-273c1299f6f1f51ad537cde49c939415R101)

Reply:
Ah, that makes sense! what about wrapping the onboarding twice? so you have an onboarding that exports onComplete and then an `onboarding-screen` that interacts with the nav? I'm not sure if that's better but it's closer to how the rest of nav calls work atm

Reply:
Oh maybe I’ve missed that. Let’s talk about this too 🤘